### PR TITLE
isort 5.13.2

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-4934](https://anaconda.atlassian.net/browse/PKG-4934)
- [Upstream repository](https://github.com/PyCQA/isort/tree/5.13.2)
- [Upstream changelog/diff](https://github.com/PyCQA/isort/blob/5.13.2/CHANGELOG.md)

### Explanation of changes:

- Set version and sha256
- Remove noarch
- Small test cleanup
- Linter fixes
- Added upstream tests

[PKG-4934]: https://anaconda.atlassian.net/browse/PKG-4934?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,34 +1,33 @@
...
   entry_points:
     - isort = isort.main:main
 
 requirements:
   host:
-    - python >=3.6.1,<4.0
+    - python
     - pip
     - wheel
     - setuptools
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
Not required, as the build is done using poetry-core.
https://github.com/PyCQA/isort/blob/5.13.2/pyproject.toml#L136-L138

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
     - poetry-core >=1.0.0
   run:
-    - python >=3.6.1,<4.0
+    - python
```
Comment (cbouss) [inline recipe/meta.yaml:24]:
```suggestion
    - python
  run_constrained:
    - colorama >=0.4.6
```
https://github.com/PyCQA/isort/blob/5.13.2/pyproject.toml#L44

**Comment by ViridianMelody** [comment]
Generally we want to run the unit tests. I took a quick look through the tests to track down which dev dependencies are imported. We don't have `pylama` so we'll need to skip that test file. The snippet below should get you most of the way there, though my local build had a few failing tests you'd need to investigate.

```
test:
  source_files:
    - tests/unit
  imports:
    - isort
  commands:
    - pip check
    - pytest tests/unit --ignore="tests/unit/test_pylama_isort.py"
  requires:
    - pip
    - pytest
    - py >=1.11.0
    - hypothesis >=6.10.1
    - colorama >=0.4.6
    - black >=22.6.0
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
 
 test:
   imports:
     - isort
-    - isort._future
   commands:
     - pip check
     - isort --help
```
Comment (cbouss) [inline recipe/meta.yaml:49]: Based on this: https://github.com/PyCQA/isort/blob/5.13.2/scripts/test.sh
You could try:
`pytest tests/unit/ -s --ignore=tests/unit/test_deprecated_finders.py`
With source_files `tests/unit/`
And required dependencies a subset of https://github.com/PyCQA/isort/blob/5.13.2/poetry.lock#L35

**Review by danpetry — APPROVED** [review]
LGTM after comments addressed and builds pass

**Review by cbouss — APPROVED** [review]

**Pull request merged**